addon/components/dynamic-form.js

```diff
@@ -15,20 +15,28 @@ const TYPE_MAP = {
 const DynamicForm = Ember.Component.extend({
 
   renderSchema: Ember.K,
+  _renderer: null,
 
-  didInsertElement() {
-    this._super(...arguments);
-    let container = getOwner(this);
-    let renderer;
-    let config = container.resolveRegistration('config:environment');
-    if (config.dynamicForms && config.dynamicForms.renderer) {
-      renderer = container.lookup(`${config.dynamicForms.renderer}:renderers`);
-    } else {
-      renderer = container.lookup('alpaca:renderers');
+  _render() {
+    var renderer = this.get('_renderer');
+    if (!renderer) {
+      let container = getOwner(this);
+      let config = container.resolveRegistration('config:environment');
+      if (config.dynamicForms && config.dynamicForms.renderer) {
+	renderer = container.lookup(`${config.dynamicForms.renderer}:renderers`);
+      } else {
+	renderer = container.lookup('alpaca:renderers');
+      }
+      this.set('_renderer', renderer);
     }
     renderer.render(this.get('renderSchema'), this.$());
   },
 
+  didInsertElement() {
```

Comment:
didReceiveAttrs happens on every render while didInsertElement happens on first render. This could cause a double render on initial render.